hack/dci-merge.sh

```diff
@@ -0,0 +1,148 @@
+            fi
+        fi
+
+        if [[ assisted =~ $SUPPORTED_HINTS ]] && grep -qi "^\s*Test-Hints:\s*assisted\s*" <<< "$DESC"; then
+            VIRT=--assisted
+            if [ -n "$ASSISTED_DCI_QUEUE" ]; then
+                DCI_QUEUE="$ASSISTED_DCI_QUEUE"
+            fi
+        fi
+
+        if [[ assisted-abi =~ $SUPPORTED_HINTS ]] && grep -qi "^\s*Test-Hints:\s*assisted-abi\s*" <<< "$DESC"; then
+            VIRT=--assisted-abi
+            if [ -n "$ASSISTED_DCI_QUEUE" ]; then
+                DCI_QUEUE="$ASSISTED_DCI_QUEUE"
+            fi
+        fi
+
+        if [[ libvirt =~ $SUPPORTED_HINTS ]] && grep -qi "^\s*Test-Hints:\s*libvirt\s*" <<< "$DESC"; then
+            VIRT=--virt
+        fi
+
+        if [[ no-check =~ $SUPPORTED_HINTS ]] && grep -qi "^\s*Test-Hints:\s*no-check\s*" <<< "$DESC"; then
+            continue
+        fi
+
+        if [[ force-check =~ $SUPPORTED_HINTS ]] && grep -qi "^\s*Test-Hints:\s*force-check\s*" <<< "$DESC"; then
+            FORCE_CHECK=--force-check
+        fi
+
+        if [[ upgrade =~ $SUPPORTED_HINTS ]] && grep -qi "^\s*Test-Upgrade-Hints:\s*yes\s*" <<< "$DESC"; then
+            UPGRADE=--upgrade
+        
+            if [ -z "$VIRT" ]; then
+                VIRT=--virt
+            fi
+
+            # process Test-Upgrade-Args-Hints
+            if [[ upgrade-args =~ $SUPPORTED_HINTS ]] && grep -qi "^\s*Test-Upgrade-Args-Hints:" <<< "$DESC"; then
+                UPGRADE_ARGS="$(sed -n -e 's/^\s*Test-Upgrade-Args-Hints:\s*//pi' <<< "$DESC")"
+            fi
+
+            # process Test-Upgrade-From-Topic-Hints
+            if [[ upgrade-from-topic =~ $SUPPORTED_HINTS ]] && grep -qi "^\s*Test-Upgrade-From-Topic-Hints:" <<< "$DESC"; then
+                UPGRADE="$UPGRADE --from-topic $(sed -n -e 's/^\s*Test-Upgrade-From-Topic-Hints:\s*//pi' <<< "$DESC")"
+            fi
+
+            # process Test-Upgrade-To-Topic-Hints
+            if [[ upgrade-to-topic =~ $SUPPORTED_HINTS ]] && grep -qi "^\s*Test-Upgrade-To-Topic-Hints:" <<< "$DESC"; then
+                UPGRADE="$UPGRADE --to-topic $(sed -n -e 's/^\s*Test-Upgrade-To-Topic-Hints:\s*//pi' <<< "$DESC")"
+            fi
+        fi
+
+        if [ -n "$VIRT" ]; then
+            # process Test-Args-Hints
+            if [[ args =~ $SUPPORTED_HINTS ]]; then
+                OPTS=$(sed -n -e "s/^\s*Test-Args-Hints:\s*//pi" <<< "$DESC")
+            else
+                OPTS=
+            fi
+
+            # process Test-App-Hints
+            if [[ app =~ $SUPPORTED_HINTS ]]; then
+                APP_NAME=$(sed -n -e "s/^\s*Test-App-Hints:\s*//pi" <<< "$DESC")
+            fi
+
+            # process Test-App-Args-Hints
+            if [[ app-args =~ $SUPPORTED_HINTS ]]; then
+                APP_ARGS=$(sed -n -e "s/^\s*Test-App-Args-Hints:\s*//pi" <<< "$DESC")
+            fi
+
+            # stop at the first valid Test-Hints: string
+            break
+        fi
+    else
+        break
+    fi
+    COMMIT="${COMMIT}^"
```

Comment:
If using in the top `git log --merges` we won't need this, and we reduce the odds to have a non-merge commit here